### PR TITLE
Support Gru, Sigmoid, Tanh ops

### PR DESCRIPTION
**Pull request by Honry**

### CODE REVIEW

**Comment by Honry** [comment]
@huningxin, PTAL, thanks!

File: onnxruntime/core/providers/webnn/builders/impl/gru_op_builder.cc [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    }
+
+    // linear_before_reset = true is not supported in WebNN at present
+    const auto linear_before_reset = helper.Get("linear_before_reset", static_cast<int32_t>(0));
+    if (linear_before_reset) {
+      LOGS(logger, VERBOSE) << "Gru unsupported linear_before_reset = true.";
+      return false;
+    }
+
+    // 'layout = rzn' is not supported in WebNN at present
```
Comment (huningxin) [inline onnxruntime/core/providers/webnn/builders/impl/gru_op_builder.cc]: "rzn" is supported https://www.w3.org/TR/webnn/#api-mlgraphbuilder-gru

Comment (Honry) [reply onnxruntime/core/providers/webnn/builders/impl/gru_op_builder.cc]: I meant webnn-native doesn't support 'rzn' at present, shell we just ignore this check and shift it off to webnn-native?

Comment (huningxin) [reply onnxruntime/core/providers/webnn/builders/impl/gru_op_builder.cc]: Yes, please do that.

Comment (Honry) [reply onnxruntime/core/providers/webnn/builders/impl/gru_op_builder.cc]: Done.

**Pull request merged**